File: Shared/src/main/java/com/lenta/shared/fmp/resources/slow/ZfmpUtz48V001.java
package com.lenta.shared.fmp.resources.slow;

import com.google.gson.annotations.SerializedName;
import com.mobrun.plugin.api.HyperHive;
import com.mobrun.plugin.api.helper.LocalTableResourceHelper;
import com.mobrun.plugin.api.request_assistant.CustomParameter;
import com.mobrun.plugin.api.request_assistant.ForbiddenScalarParameter;
import com.mobrun.plugin.api.request_assistant.RequestBuilder;
import com.mobrun.plugin.models.StatusSelectTable;

public class ZfmpUtz48V001 {

    public static final String NAME_RESOURCE = "ZFMP_UTZ_48_V002";
    public static final String NAME_OUT_PARAM_ET_MATNR_LIST = "ET_MATNR_LIST";
    public static final String LIFE_TIME = "1 day, 0:00:00";

    private final HyperHive hyperHive;

    public final LocalTableResourceHelper<ItemLocal_ET_MATNR_LIST, Status_ET_MATNR_LIST> localHelper_ET_MATNR_LIST;


    public ZfmpUtz48V001(HyperHive hyperHive) {

        this.hyperHive = hyperHive;

        localHelper_ET_MATNR_LIST = 
                 new LocalTableResourceHelper<ItemLocal_ET_MATNR_LIST, Status_ET_MATNR_LIST>(NAME_RESOURCE, 
                         NAME_OUT_PARAM_ET_MATNR_LIST, 
                         hyperHive,
                         Status_ET_MATNR_LIST.class);

    }

    public RequestBuilder<Params, ForbiddenScalarParameter> newRequest() { return new RequestBuilder<Params, ForbiddenScalarParameter>(hyperHive, NAME_RESOURCE, true);}

    static final class Status_ET_MATNR_LIST extends StatusSelectTable<ItemLocal_ET_MATNR_LIST> {}

    public static class ItemLocal_ET_MATNR_LIST {
        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("MATERIAL")
        public String material;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("NAME")
        public String name;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("MATYPE")
        public String matype;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("MATCODE")
        public String matcode;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("BUOM")
        public String buom;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("BSTME")
        public String bstme;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("MATR_TYPE")
        public String matrType;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("ABTNR")
        public String abtnr;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("IS_RETURN")
        public String isReturn;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("IS_EXC")
        public String isExc;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("IS_ALCO")
        public String isAlco;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("IS_SET")
        public String isSet;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("MATKL")
        public String matkl;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("EKGRP")
        public String ekgrp;

        //  type: BIGINT, source: {'name': 'SAP', 'type': 'I'}
        @SerializedName("MHDHB_DAYS")
        public Integer mhdhbDays;

        //  type: BIGINT, source: {'name': 'SAP', 'type': 'I'}
        @SerializedName("MHDRZ_DAYS")
        public Integer mhdrzDays;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("IS_MARK")
        public String isMark;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("IS_VET")
        public String isVet;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("IS_HF")
        public String isHf;

        //  type: TEXT, source: {'name': 'SAP', 'type': 'C'}
        @SerializedName("IS_NEW")
        public String isNew;


    }


    public interface Params extends CustomParameter {}


}

File: work_list/src/main/java/com/lenta/bp14/platform/extentions/ZfmpUtz48V001.kt
package com.lenta.bp14.platform.extentions

import com.lenta.shared.fmp.resources.slow.ZfmpUtz48V001

fun ZfmpUtz48V001.ItemLocal_ET_MATNR_LIST.toCheckListGoodInfo(): CheckListGoodInfo? {
    return if (material == null && name == null && buom == null) {
        null
    } else {
        CheckListGoodInfo(
                material = material.orEmpty(),
                name = name.orEmpty(),
                unitsCode = buom.orEmpty()
        )
    }
}

fun ZfmpUtz48V001.ItemLocal_ET_MATNR_LIST.toWorkListGoodInfo(): WorkListGoodInfo? {
    return if (material == null && name == null && buom == null) {
        null
    } else {
        return WorkListGoodInfo(
                material = material.orEmpty(),
                name = name.orEmpty(),
                unitsCode = buom.orEmpty(),
                goodGroup = matkl.orEmpty(),
                purchaseGroup = ekgrp.orEmpty(),
                shelfLife = mhdhbDays ?: 0,
                remainingShelfLife = mhdrzDays ?: 0,
                matrixType = matrType.orEmpty(),
                section = abtnr.orEmpty(),
                isExcise = isExc.orEmpty(),
                isAlcohol = isAlco.orEmpty(),
                isMark = isMark.orEmpty(),
                isRusWine = isRusWine,
                healthFood = isHf.orEmpty(),
                novelty = isNew.orEmpty()
        )
    }
}

data class CheckListGoodInfo(
        val material: String,
        val name: String,
        val unitsCode: String
)

data class WorkListGoodInfo(
        val material: String,
        val name: String,
        val unitsCode: String,
        var goodGroup: String,
        var purchaseGroup: String,
        val shelfLife: Int,
        val remainingShelfLife: Int,
        val matrixType: String,
        val section: String,
        val isExcise: String,
        val isAlcohol: String,
        val isRusWine: String?, // todo Что это такое, оно реально используется?
        val isMark: String,
        val healthFood: String,
        val novelty: String
)